### PR TITLE
dm: add strict optimist mode

### LINKED ISSUE #9112: dm: add strict optimistic shard mode
### Is your feature request related to a problem?

For some DDL like `alter table rename column`, we skip all dml and ddl for that shard table until all shard tables meet the same rename column statement in optimistic mode. However, some user use optimistic mode in test environment and want to know the incompatible DDL immediately.

### Describe the feature you'd like

add a config item `strict-optimistic-shard-mode', when we meet DDL like `alter table rename column`, return the error immediately.

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close #9112 

### What is changed and how it works?
- return error when get SkipAndWaitRedirect operator.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
support strict optimistic mode for incompatible DDL
```

### CODE REVIEW

File: dm/config/subtask.go [hunk 1]
```diff
@@ -84,9 +84,10 @@ type SubTaskConfig struct {
 	flagSet *flag.FlagSet
 
 	// when in sharding, multi dm-workers do one task
-	IsSharding bool   `toml:"is-sharding" json:"is-sharding"`
-	ShardMode  string `toml:"shard-mode" json:"shard-mode"`
-	OnlineDDL  bool   `toml:"online-ddl" json:"online-ddl"`
+	IsSharding                bool   `toml:"is-sharding" json:"is-sharding"`
+	ShardMode                 string `toml:"shard-mode" json:"shard-mode"`
+	StrictOptimisticShardMode bool   `toml:"strict-optimistic-shard-mode" json:"strict-optimistic-shard-mode"`
```
Comment (lance6716) [inline dm/config/subtask.go:89]: can we move it as a value of `shard-mode`?

Comment (GMHDBJD) [reply dm/config/subtask.go:89]: have discuss with pm. the result is new-value is better

**Review by lance6716 — APPROVED** [review]

**Pull request merged**